nemo_skills/training/data_preparation_utils/preprocessing.py

```diff
@@ -240,31 +240,46 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.prompt_config = prompt_config
-        self.prompt_template = prompt_template
         self.input_key = input_key
         self.output_key = output_key
         self.chat_format = chat_format
         self.metadata = metadata
         self.exclude_optional_keys = exclude_optional_keys
-        self.generation_suffix = generation_suffix
-        if self.generation_suffix and self.chat_format:
-            raise ValueError("generation_suffix can only be used with chat_format=False")
-        if self.prompt_config is None:
-            raise ValueError("`prompt_config` should be provided")
-        if self.prompt_template is None:
-            raise ValueError("`prompt_template` should be provided")
         if not self.metadata:
             self.metadata = {}
 
+        self.prompt = None
+        if prompt_config and prompt_template:
+            self.prompt = get_prompt(prompt_config, prompt_template)
+        else:
+            LOG.warning("Prompt details are missing! The processed data won't be formatted using any prompt.")
+
+        if self.chat_format:
+            if generation_suffix:
```

Comment:
tbh, I'd just remove it as a parameter completely and always add assistant end if it's non-chat format and prompt is provided. I don't see any case where we wouldn't want to do that or would need to customize

Reply:
Sure, let me change it.